pkg/apis/certmanager/validation/certificate_for_issuer.go

```diff
@@ -51,8 +51,8 @@ func ValidateCertificateForIssuer(crt *v1alpha1.Certificate, issuerObj v1alpha1.
 func ValidateCertificateForACMEIssuer(crt *v1alpha1.CertificateSpec, issuer *v1alpha1.IssuerSpec, specPath *field.Path) field.ErrorList {
 	el := field.ErrorList{}
 
-	if crt.KeyAlgorithm != v1alpha1.KeyAlgorithm("") && crt.KeyAlgorithm != v1alpha1.RSAKeyAlgorithm {
-		el = append(el, field.Invalid(specPath.Child("keyAlgorithm"), crt.KeyAlgorithm, "ACME key algorithm must be RSA"))
+	if crt.KeyAlgorithm != v1alpha1.KeyAlgorithm("") && crt.KeyAlgorithm != v1alpha1.RSAKeyAlgorithm && crt.KeyAlgorithm != v1alpha1.ECDSAKeyAlgorithm {
+		el = append(el, field.Invalid(specPath.Child("keyAlgorithm"), crt.KeyAlgorithm, "ACME key algorithm must be RSA or ECDSA"))
```

Comment:
Can your remove this section of code altogether? We have the generic `ValidateCertificate` function which will verify that the keyAlgorithm is one of RSA or ECDSA. This function is *only* called for ACME certificates, and this segment will mean that entering an invalid keyAlgorithm on an ACME cert will generate *two* errors 😄

Reply:
done